xrootd/client/auth.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2018 The go-hep Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package client // import "go-hep.org/x/hep/xrootd/client"
+
+import (
+	"bytes"
+	"context"
+	"os/user"
+
+	"github.com/pkg/errors"
+	"go-hep.org/x/hep/xrootd/xrdproto/auth"
+)
+
+func (client *Client) auth(ctx context.Context, securityInformation []byte) error {
+	securityInformation = bytes.TrimLeft(securityInformation, "&")
+	securityProviders := bytes.Split(securityInformation, []byte{'&'})
+
+	var errs []error
+	for _, securityProvider := range securityProviders {
+		securityProvider = bytes.TrimLeft(securityProvider, "P=")
+		if bytes.Equal(securityProvider, auth.UnixType[:]) {
+			u, err := user.Current()
+			if err != nil {
+				errs = append(errs, errors.WithMessage(err, "xrootd: could not authorize using unix"))
+				continue
+			}
+			var group string
+			g, err := user.LookupGroupId(u.Gid)
+			if err != nil {
```

Comment:
rather than this relatively _ad hoc_ handling, perhaps we could create a little helper function that forwards to `os/user.LookupGroupId` when on unix and do what you did below when on windows.

that could perhaps look something like (for `lookupgid_unix.go`):
```go
//+build !windows

func lookupGroupID(usr *user.User) (string, error) { ... }
```
and (for `lookupgid_windows.go`):
```go
//+build windows

func lookupGroupID(usr *user.User) (string, error) { ... }
```